lib/Matrix.js

```diff
@@ -44,6 +45,19 @@ function useMatrixProvider(auth) {
 
         const roomState = room.getLiveTimeline().getState(EventTimeline.FORWARDS);
 
+        // check if any users have knocked on the room
+        const knockingMembers = room.getMembersWithMembership('knock');
```

Comment:
This is kind of a bad place for this logic, because now we're just shuffling it down somewhere inside of `buildRoomObject`. It's bad for multiple reasons:

1. `buildRoomObject` is now doing more than what its name says it does; it's updating our internal state, so this becomes an undocumented (and imo unwanted) side-effect
2. At which times exactly do we call this function `buildRoomObject()`? I'd say we don't really know from the top of our heads; so by throwing in the logic to keep our pending knocks up to date, it looks more like: "ah, whatever... this seems to kind of work, let's just put it down somewhere in here"... this leads to:
3. Let's think -when- exactly Matrix' sync call tells us about changes in knocks. I think that it's always a `Room.timeline` event... so why don't we add the logic in there?
4. (Most likely every `Room.timeline` event / each `RoomTimelineEvent()` call is going to be for -one- pending knock?! I'm just guessing here... but this might mean that it should be fairly easy to just take whatever info is inside `event` and push it into our state via `setKnockingMembers`...)